redis/asyncio/connection.py

```diff
@@ -730,6 +730,27 @@ def _add_to_local_cache(
         ):
             self.client_cache.set(command, response, keys)
 
+    def flush_cache(self):
+        try:
+            if self.client_cache:
```

Comment:
Why do you need the `if` inside the `try` block?

Reply:
The `try` is meant to compensate for the `if`? You mean we could leave the `if` without the `try`? I would find that cleaner.

Reply:
The if-statement will prevent the call to flush/delete_command/invalidate_key that would cause the AttributeError if client_cache is falsy. The try-catch-block will also catch an AttributeError if client_cache is truthy but doesn't provide the method.
I'm missing to much context to say which is more desirable. But if it can be assumed, that a thruthy client_cache always also provides the methods, then the if-statement and try-catch are redundant and it becomes a question of style. You can read more on that here https://realpython.com/python-lbyl-vs-eafp/
Best regards, a random passerby XD

Reply:
Thanks @CBerndt-Work for your nice explanation. The article is very well written. I realized that I am by now too conditioned towards LBYL, from other languages. Especially here, using `AttributeError`, which signals issues with the code structure, not with data. So I went for the `if`s.